sphinx/util/docutils.py

```diff
@@ -479,8 +479,8 @@ def dispatch_visit(self, node):
         for node_class in node.__class__.__mro__:
             method = getattr(self, 'visit_%s' % (node_class.__name__), None)
             if method:
-                logger.debug('SphinxTranslator.dispatch_visit calling %s for %s' %
-                             (method.__name__, node))
```

Comment:
Before this change, this line goes to format message before calling `logger.debug()` even not in debug mode. And it sometime causes large node-to-string conversion always. Because `%` operator is stronger than method calling.

After the change, the conversion will go only in debug mode. So we should give parts of messages via parameters of `logger.debug()`.

Reply:
This is a subtle find...